File: 5_semester/DSP/lab3/lib/logic/variant.dart
import 'dart:math';

import 'package:extend_math/extend_math.dart';

abstract final class Variant {
  static const n = 22;
  static const f = 3 * n;
  static const t = 10 / f;
  static const interval = MathInterval(-t, t);
  static const step = 1 / 3 / f;
  static final fxDots = interval.applyFx(fx, step: step);

  static double fx(double x) => sin(2 * pi * f * x);
}

File: 5_semester/ЦОПС/lab2/variant.py
import math as math
import numpy as np

n = 10

fn = 3 * n
t = 10 / fn
step = 0.001
fx = lambda x: math.sin(2 * math.pi * fn * x)

interval = np.arange(0, t, step)
fx_dots = np.vectorize(fx)(interval)
